tests/tests_assemble_workflow/test_dist.py

```diff
@@ -4,21 +4,63 @@
-    def test_from_path_tar_gz(self) -> None:
-        dist = Dist.from_path("opensearch", "filename.tar.gz", "opensearch-1.1.0")
-        self.assertIs(type(dist), DistTar)
-
-    def test_from_path_zip(self) -> None:
-        dist = Dist.from_path("opensearch", "filename.zip", "opensearch-1.1.0")
-        self.assertIs(type(dist), DistZip)
-
-    def test_from_path_invalid(self) -> None:
-        with self.assertRaises(ValueError) as ctx:
-            Dist.from_path("opensearch", "filename.invalid", "opensearch-1.1.0")
-        self.assertEqual(str(ctx.exception), 'Invalid min "dist" extension in input artifacts: .invalid (filename.invalid).')
+
+    def setUp(self) -> None:
+
+        self.artifacts_path = os.path.join(os.path.dirname(__file__), "data/artifacts/dist/")
+        self.distTar = DistTar(
+            "OpenSearch",
+            self.artifacts_path + "opensearch-min-1.3.0-linux-x64.tar.gz",
+            "opensearch-1.3.0"
+        )
+
+    def test_dist_variables(self) -> None:
+        self.assertEqual(self.distTar.name, "OpenSearch")
+        self.assertEqual(os.path.exists(self.distTar.path), True)
+        self.assertEqual(self.distTar.min_path, "opensearch-1.3.0")
+
+    @patch("assemble_workflow.dist.Dist.find_min_archive_path", return_value="opensearch-1.3.0")
+    @patch("assemble_workflow.dist.Dist.rename_archive_path", return_value="test_path")
+    @patch("assemble_workflow.dist.DistTar.__extract__")
+    def test_dist_extract(self, dist_find_min_path: Mock, dist_rename_path: Mock, distTar_extract: Mock) -> None:
+        archive_path = self.distTar.extract("test_dest")
+        self.assertEqual(archive_path, "test_path")
+        dist_find_min_path.assert_called_once()
+        dist_rename_path.assert_called_once()
+        distTar_extract.assert_called_once()
+
+    @patch("assemble_workflow.dist.DistTar.__build__")
+    @patch("shutil.copyfile")
+    def test_dist_build(self, distTar_build: Mock, shutil_copyfile: Mock) -> None:
+        self.distTar.build("temp_name", "temp_dest")
+        distTar_build.assert_called_once()
+        shutil_copyfile.assert_called_once()
+
+    def test_find_min_archive_path(self) -> None:
+        self.assertEqual(
+            self.distTar.find_min_archive_path(self.artifacts_path),
+            self.artifacts_path + "opensearch-1.3.0"
+        )
+
+    @patch("os.path.basename", return_value="opensearch-1.3.0")
+    def test_rename_archive_path_norename(self, os_path_basename: Mock) -> None:
+        self.assertEqual(
+            self.distTar.rename_archive_path("temp_path/opensearch-1.3.0"),
+            "temp_path/opensearch-1.3.0"
+        )
+
+    @patch("os.path.dirname", return_value="temp_path")
+    @patch("os.rename")
+    def test_rename_archive_path_rename(self, os_path_dirname: Mock, os_rename: Mock) -> None:
+        self.assertEqual(
+            self.distTar.rename_archive_path("temp_path/opensearch-x.y.z"),
+            "temp_path/opensearch-1.3.0"
```

Comment:
It seems like this is failing the python check for windows because of the path format.

Reply:
fixed.